ui/components/data-sources/DataSourcesList.js

```diff
@@ -1,11 +1,48 @@
 import React from "react";
 import { Query } from "react-apollo";
-import { ListView } from "patternfly-react";
+import {
+    ListView,
+    Button,
+    EmptyState,
+    EmptyStateIcon,
+    EmptyStateTitle,
+    EmptyStateInfo,
+    EmptyStateHelp,
+    EmptyStateAction
+} from "patternfly-react";
 
 import GetDataSources from "../../graphql/GetDataSources.graphql";
 import { DataSourcesListItem } from "./DataSourcesListItem";
 
-const DataSourcesList = ({ filter }) => (
+const EmptyList = ({ action }) => (
```

Comment:
I think this one should be in a different file, better keep components small for maintainability